sematic/api_client.py

```diff
@@ -72,13 +70,52 @@ def get_artifact_value_by_id(
     Any
         The value of the requiested artifact.
     """
-    # TODO: Store storage type on artifact
-    if storage is None:
-        storage = S3Storage()
-
     artifact = _get_artifact(artifact_id)
 
-    return get_artifact_value(artifact, storage)
+    return get_artifact_value(artifact)
+
+
+def get_artifact_value(artifact: Artifact) -> Any:
+    storage_engine, location = _get_artifact_location(artifact.id, mode=StorageMode.READ)
+    payload = storage_engine.get(location)
+
+    return deserialize_artifact_value(artifact, payload)
+
+
+def store_artifact_value(artifact_id: str, value: bytes):
+    storage_engine, location = _get_artifact_location(artifact_id, StorageMode.WRITE)
+    storage_engine.set(location, value)
+
+
+def _get_artifact_location(artifact_id: str, mode: StorageMode) -> Tuple[Storage, str]:
+    response = _get(f"/artifacts/{artifact_id}/location/{mode.value}")
+
+    return _get_location_for_response(response)
+
+
+def store_future_pickle(future_id: str, pickle: bytes):
+    storage_engine, location = _get_future_location(future_id, StorageMode.WRITE)
+    storage_engine.set(location, pickle)
+
+
+def get_future_pickle(future_id: str) -> bytes:
+    storage_engine, location = _get_future_location(future_id, StorageMode.READ)
+    return storage_engine.get(location)
+
+
+def _get_future_location(future_id: str, mode: StorageMode) -> Tuple[Storage, str]:
+    response = _get(f"/runs/{future_id}/location/{mode.value}")
+
+    return _get_location_for_response(response)
+
+
+def _get_location_for_response(response: Dict[str, str]) -> Tuple[Storage, str]:
+    try:
+        storage_engine = get_storage(StorageSettingValue[response["storage_engine"]])
+    except KeyError:
+        raise NotImplementedError(f"Unknown storage engine: {response['storage_engine']}")
+
+    return storage_engine(), response["location"]
```

Comment:
After implementing the previous comment, this will be:
```suggestion
    storage_engine = get_storage(StorageSettingValue[response["storage_engine"]])

    return storage_engine(), response["location"]
```